packages/aws-cdk/lib/notices.ts

```diff
@@ -0,0 +1,270 @@
+import * as https from 'https';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import * as semver from 'semver';
+import { print, debug } from './logging';
+import { cdkCacheDir } from './util/directories';
+import { versionNumber } from './version';
+
+const CACHE_FILE_PATH = path.join(cdkCacheDir(), 'notices.json');
+
+export interface DisplayNoticesProps {
+  /**
+   * The cloud assembly directory. Usually 'cdk.out'.
+   */
+  readonly outdir: string;
+
+  /**
+   * Issue numbers of notices that have been acknowledged by a user
+   * of the current CDK repository. These notices will be skipped.
+   */
+  readonly acknowledgedIssueNumbers: number[];
+
+  /**
+   * Setting that comes from the command-line option.
+   * For example, `cdk synth --no-notices`.
+   */
+  readonly temporarilySuppressed?: boolean;
+
+  /**
+   * Setting that comes from cdk.json. For example,
+   *
+   * "context": {
+   *   "notices": false
+   * }
+   */
+  readonly permanentlySuppressed?: boolean;
```

Comment:
The setting in `context` is just another way of passing `--no-notices`.

1. No reason to call this function in the first place
2. No need to treat `{ "notices": false }` different from `--no-notices`.

By the way, `{ "notices": false }` should be at the top-level in `cdk.json`, not in the `context` dictionary. It's CLI config, not app context.